adr/0006-reviewer-responsibilies.adoc

```diff
@@ -0,0 +1,74 @@
+
+== Decision
+
+There are three levels of reviewing.
+
+. Any one can review a Pull Request (PR) and give constructive feedback on different areas (code, documentation, feature experience, security, etc.).
+  This is an optional but recommended step.
+  More eyes means higher quality.
+. A PR is reviewed by a _binding reviewer_ who is fluent in the area being touched: this person is responsible for signing with blood of the usefulness, quality of the code and soundness of the solution.
+  That’s the review of the expert and the most critical one for quality.
+  This review is mandatory.
+.. Unfortunately, we do not always have (designed or effective) experts in all areas so it’s a constant work in progress and might make this step optional in some situations.
+. A PR is reviewed by a committer (push rights) whose responsibilities are:
+.. Do a general review (often from a non expert eye).
+.. Ensure consistency and quality across Quarkus’s codebase (code, concepts, API consistency, documentation).
+.. Detect potential security concerns with an attacker mindset (though not being an actual security professional).
+.. Ensure consistency at the platform level (duplicated or overlapping features, usage integration issues across features etc).
+.. Effectively push the code to the main branch.
+
+=== TODO Open questions to resolve before closing
+
+How is a person moving from casual reviewer to binding reviewer?
+People will tend to request review from binding reviewers leading to a progression problem where casual reviewers will never have the knowledge to be “promoted”.
+We have a few people with no specific area of ownership, that’s specifically acute there.
+
+How can a non binding review see the value of its work considering someone else has to go behind them anyways.
+Feels like double work where the team is very busy.
+
+How can we define SLO for the PR review to help with the _flow_, satisfy contributors while keeping quality high.
+And how to use it as a mechanism to improve and know where to get help?
+The faster the flow, the high an organisation measure in effectiveness, happiness etc.
+
+==== Proposition 1
+
+Define and measure (GitHub APIs) SLO for the time to push or drop / withdraw.
+This is not a perfect proxy as some PRs do demand multi weeks conversations but measuring and dashboarding is our first step to know in which area we have a problem.
+
+We could define a team intended SLO (say 3 weeks to start but that’s already quite long) and know which areas chronically fall behind.
+
+We can introduce the notion of binding reviewer leutenants (like in the Linux kernel) where a binding reviewer whose area is not meeting the SLO would train other peoples (actively delegating the PR work as a first phase before its own review, mentoring, answering questions, etc).
```

Comment:
```suggestion
We can introduce the notion of binding reviewer lieutenants (like in the Linux kernel), where a binding reviewer whose area is not meeting the SLO would train other people (actively delegating the PR work as a first phase before its own review, mentoring, answering questions, etc.).
```

Reply:
I like the idea, but I'm not sure of the implementation. For me, even lieutenants would require a set of contributions before being named lieutenants and if they do that, they could become binding reviewers. Otherwise, it would mean that the last review (the merge part) would need to do a more extensive review (which would add even more latency)